client/src/App.tsx

```diff
@@ -52,22 +51,35 @@ function OAuth() {
 function App() {
   const api = new Api();
   const navigate = useNavigate();
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    const validateSession = async () => {
-      const res = await api.validateSession()
-      if (res?.redirect) {
+    const refreshToken = async () => {
```

Comment:
the problem here is that if the refresh token exists, the API will never throw a 401 error because every time the access_token expires, the googleapi sdk will automatically obtain the new token using the refresh token but won't update the jwt payload with the new token. 

However, as you suggested, we can fix this, by not supplying the refresh_token in the OAuth intercepter in the first place. So, if the `access_token` expires, it will throw a 401 error, and then the client can request to refresh the token.